packages/mediapipe-core/lib/src/task_options.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:typed_data';
+import 'package:equatable/equatable.dart';
+import 'mediapipe_common_bindings.dart' as bindings;
+import 'ffi_utils.dart';
+
+class BaseOptions extends Equatable {
+  const BaseOptions({this.modelAssetBuffer, this.modelAssetPath})
+      : assert(
+          !(modelAssetBuffer == null && modelAssetPath == null),
+          'You must supply either `modelAssetBuffer` or `modelAssetPath`',
+        ),
+        assert(
+          !(modelAssetBuffer != null && modelAssetPath != null),
```

Comment:
And this ensures both are not provided.

Might be worth unpacking both into a single assertion with a function body so you could use control flow for better readability and error messaging.